control-plane/stor-port/src/types/v0/store/volume.rs

```diff
@@ -172,10 +172,44 @@ pub struct VolumeSpec {
     /// Affinity Group related information.
     #[serde(default)]
     pub affinity_group: Option<AffinityGroup>,
-    /// Runtime volume information, useful to quick checks without having to read out from PSTOR
+    /// Volume metadata information.
+    #[serde(default)]
+    pub metadata: VolumeMetadata,
+}
+
+/// Volume meta information.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default)]
+pub struct VolumeMetadata {
+    /// Persisted metadata information.
+    persisted: VolumePersistedMetadata,
+    /// Runtime information, useful to quick checks without having to read out from PSTOR
     /// or any other control-plane related registry.
     #[serde(skip)]
-    pub runtime_info: VolumeRuntimeMetadata,
+    runtime: VolumeRuntimeMetadata,
+}
+impl VolumeMetadata {
+    /// Create a new `Self` from the given parameters.
+    pub fn new(as_thin: bool) -> Self {
+        Self {
+            persisted: VolumePersistedMetadata {
+                snapshot_as_thin: as_thin,
+            },
+            runtime: Default::default(),
+        }
+    }
+    /// Insert snapshot in the list.
+    fn insert_snapshot(&mut self, snapshot: SnapshotId) {
+        self.runtime.snapshots.insert(snapshot);
+        // we become thin provisioned!
+        self.persisted.snapshot_as_thin = true;
```

Comment:
I think we'll be doing this for each snapshot creation but effectively only first snapshot is only doing this? Shall we make it more intuitive like:
```
if !self.persisted.snapshot_as_thin
    self.persisted.snapshot_as_thin = true;
```

Reply:
hmm I don't see that as more intuitive, it's less to me before it's redundant.